src/authInfo.ts

```diff
@@ -1085,12 +1087,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     try {
       this.logger.info(`Sending request for Username after successful auth code exchange to URL: ${url}`);
       const response = await new Transport().httpRequest({ url, headers });
-      return asString(parseJsonMap(response.body).Username);
+      if (response.statusCode >= 400) {
+        this.throwUserGetException(response as never);
+      } else {
+        return asString(parseJsonMap(response.body).Username);
+      }
     } catch (err) {
       throw SfdxError.create('@salesforce/core', 'core', 'AuthCodeUsernameRetrievalError', [orgId, err.message]);
     }
   }
 
+  /**
+   * Given an error while getting the User object, handle different possibilities of response.body.
+   *
+   * @param response
+   * @private
+   */
+  private throwUserGetException(response: never) {
+    let messages = '';
+    let body: AnyJson | string = getString(
```

Comment:
extract body from response, if not present provide a parsable string that represents an UNKNOWN failure

Reply:
NIT: I would prefer to have two different variables rather than sharing a type. Maybe `const possibleJsonBody` here and `const body` on line 1114?